modules/dag-l0/src/main/scala/org/tessellation/dag/l0/domain/snapshot/programs/Download.scala

```diff
@@ -121,23 +120,30 @@ object Download {
         val (lastSnapshot, lastContext) = result
 
         fetchSnapshot(none, lastSnapshot.ordinal.next).flatMap { snapshot =>
-          lastSnapshot.toHashed[F].flatMap { hashed =>
-            Applicative[F].unlessA {
-              Validator.isNextSnapshot(hashed, snapshot.value)
-            }(InvalidChain.raiseError[F, Unit])
-          } >>
+          hasherSelector
+            .forOrdinal(lastSnapshot.ordinal) { implicit hasher =>
+              lastSnapshot.toHashed[F]
+            }
+            .flatMap { hashed =>
+              Applicative[F].unlessA {
+                Validator.isNextSnapshot(hashed, snapshot.value)
+              }(InvalidChain.raiseError[F, Unit])
+            } >>
             globalSnapshotContextFns
               .createContext(lastContext, lastSnapshot, snapshot)
               .handleErrorWith(_ => InvalidChain.raiseError[F, GlobalSnapshotContext])
               .flatTap { _ =>
                 snapshotStorage.writePersisted(snapshot)
```

Comment:
It's not clear to me which hasher is used here when the hash filename is generated.